pkg/trace/agent/agent_test.go

```diff
@@ -189,7 +189,7 @@ func TestProcess(t *testing.T) {
 
 		agnt.Process(&api.Payload{
 			TracerPayload: testutil.TracerPayloadWithChunk(testutil.TraceChunkWithSpan(spanValid)),
-			Source:        info.NewReceiverStats().GetTagStats(info.Tags{}),
+			Source:        want,
```

Comment:
The assertions `assert.EqualValues(0, want.TracesFiltered.Load())` and `assert.EqualValues(0, want.SpansFiltered.Load())` were passing because they're checking an empty tag stats (`want`) while the process function was using a different tag stats instance.

Prior to this change, the tests would technically "pass" even with `cfg.Ignore["resource"] = []string{".*"}`.